sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -157,6 +158,10 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
       case _: ShowTableExtended =>
         throw QueryCompilationErrors.commandUnsupportedInV2TableError("SHOW TABLE EXTENDED")
 
+      case operator: LogicalPlan
+        if !Utils.isTesting && operator.output.exists(_.dataType.isInstanceOf[TimestampNTZType]) =>
```

Comment:
Ah, a requirement is we should call `dataType` only on resolved attributes, so the check should be
```
operator.output.exists(attr => attr.resolved && attr.dataType.isInstanceOf[TimestampNTZType])
```

Reply:
@gengliangwang can you help to fix it?

Reply:
Yes I am about to create a PR